kornia/augmentation/_2d/geometric/shear.py

```diff
@@ -0,0 +1,112 @@
+        same_on_batch: apply the same transformation across the batch.
+        align_corners: interpolation flag.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it to the batch form (False).
+
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)`
+
+    .. note::
+        This function internally uses :func:`kornia.geometry.transform.warp_affine`.
+
+    Examples:
+        >>> import torch
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.rand(1, 1, 3, 3)
+        >>> aug = RandomShear((-5., 2., 5., 10.), p=1.)
+        >>> out = aug(input)
+        >>> out, aug.transform_matrix
+        (tensor([[[[0.4420, 0.7598, 0.1536],
+                  [0.1740, 0.3095, 0.6121],
+                  [0.4450, 0.8892, 0.4045]]]]), tensor([[[ 1.0000,  0.0100, -0.0100],
+                 [-0.1183,  0.9988,  0.1171],
+                 [ 0.0000,  0.0000,  1.0000]]]))
+        >>> aug.inverse(out)
+        tensor([[[[0.4068, 0.7549, 0.1404],
+                  [0.2051, 0.3111, 0.5562],
+                  [0.3985, 0.8816, 0.4257]]]])
+
+    To apply the exact augmenation again, you may take the advantage of the previous parameter state:
+        >>> input = torch.randn(1, 3, 32, 32)
+        >>> aug = RandomShear((-15., 20.), p=1.)
+        >>> (aug(input) == aug(input, params=aug._params)).all()
+        tensor(True)
+    """
+
+    def __init__(
+        self,
+        shear: Union[Tensor, float, Tuple[float, float], Tuple[float, float, float, float]],
+        resample: Union[str, int, Resample] = Resample.BILINEAR.name,
+        same_on_batch: bool = False,
+        align_corners: bool = False,
+        padding_mode: Union[str, int, SamplePadding] = SamplePadding.ZEROS.name,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        self._param_generator: rg.ShearGenerator = rg.ShearGenerator(shear)
+        self.flags = dict(
+            resample=Resample.get(resample), padding_mode=SamplePadding.get(padding_mode), align_corners=align_corners
+        )
+
+    def compute_transformation(self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
+        return get_shear_matrix2d(
+            torch.as_tensor(params["center"], device=input.device, dtype=input.dtype),
+            deg2rad(torch.as_tensor(params["shear_x"], device=input.device, dtype=input.dtype)),
+            deg2rad(torch.as_tensor(params["shear_y"], device=input.device, dtype=input.dtype)),
+        )
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        _, _, height, width = input.shape
+        transform = cast(Tensor, transform)
```

Comment:
`warp_affine` will raise an error if the transform is None, why cast here? Can't the typing be just tensor?

Reply:
If is because the `apply_transform` typing will diverge from the Base, we can just `# type: ignore[override]`

Reply:
I will throw an error for None values. I feel like `# type: ignore[override]` only silent the error message but not an elegant solution.

Reply:
agree, but maybe is easier to find/fix the typing on the base than remove the cast after -- at least the `ignore[override]` is the _temporary_ solution that I apply on some augmentations